lib/db.js

```diff
@@ -260,7 +265,43 @@ class LimitDBRedis extends EventEmitter {
           }
           return callback(null, res);
         });
-    })
+    });
+  }
+
+  takeExponential(params, callback) {
+    this._doTake(params, callback, (key, bucketKeyConfig, count) => {
+      this.redis.takeExponential(key,
+        bucketKeyConfig.ms_per_interval || 0,
+        bucketKeyConfig.size,
+        count,
+        Math.ceil(bucketKeyConfig.ttl || this.globalTTL),
+        bucketKeyConfig.drip_interval || 0,
+        bucketKeyConfig.backoff_factor || 2,
+        bucketKeyConfig.multiple_unit || 1000,
+        (err, results) => {
+          if (err) {
+            return callback(err);
+          }
+          const remaining = parseInt(results[0], 10);
+          const conformant = parseInt(results[1], 10) ? true : false;
+          const reset = parseInt(results[3], 10);
+          const backoff_factor = parseInt(results[4], 10);
+          const backoff_time = parseInt(results[5], 10);
+          const res = {
+            conformant,
+            remaining,
+            reset: Math.ceil(reset / 1000),
+            limit: bucketKeyConfig.size,
+            delayed: false,
+            backoff_factor,
+            backoff_time
```

Comment:
can we add the delta_reset_ms?
```suggestion
            backoff_time,
            delta_reset_ms: Math.max(reset - parseInt(results[2], 10), 0)

```